libcloud/dns/drivers/vultr.py

```diff
@@ -0,0 +1,391 @@
+
+        :param zone: Zone to list records for.
+        :type zone: :class:`Zone`
+
+        :return: ``list`` of :class:`Record`
+        """
+        action = '/v1/dns/list'
+        params = {'api_key': self.key}
+        response, errors = self.connection.request(action=action,
+                                                   params=params).parse_body()
+        zones = self._to_zones(response[0])
+
+        return zones
+
+    def list_records(self, zone):
+        """
+        Returns a list of records for the provided zone.
+
+        :param zone: zone to list records for
+        :type zone: `Zone`
+
+        :rtype: list of :class: `Record`
+        """
+        if not isinstance(zone, Zone):
+            raise ZoneRequiredException('zone should be of type Zone')
+
+        zones = self.list_zones()
+
+        if not self.ex_zone_exists(zone.domain, zones):
+            raise ZoneDoesNotExistError(value='', driver=self,
+                                        zone_id=zone.domain)
+
+        action = '/v1/dns/records'
+        params = {'domain': zone.domain}
+        response, errors = self.connection.request(action=action,
+                                                   params=params).parse_body()
+        records = self._to_records(response[0], zone=zone)
+
+        return records
+
+    def get_zone(self, zone_id):
+        """
+        Returns a `Zone` instance.
+
+        :param zone_id: name of the zone user wants to get.
+        :type zone_id: ``str``
+
+        :rtype: :class:`Zone`
+        """
+        ret_zone = None
+
+        action = '/v1/dns/list'
+        params = {'api_key': self.key}
+        response, errors = self.connection.request(action=action,
+                                                   params=params).parse_body()
+        zones = self._to_zones(response[0])
+
+        if not self.ex_zone_exists(zone_id, zones):
+            raise ZoneDoesNotExistError(value=None, zone_id=zone_id,
+                                        driver=self)
+
+        for zone in zones:
+            if zone_id == zone.domain:
+                ret_zone = zone
+
+        return ret_zone
+
+    def get_record(self, zone_id, record_id):
+        """
+        Returns a Record instance.
+
+        :param zone_id: name of the required zone
+        :type zone_id: ``str``
+
+        :param record_id: ID of the required record
+        :type record_id: ``str``
+
+        :rtype: :class: `Record`
+        """
+        ret_record = None
+        zone = self.get_zone(zone_id=zone_id)
+        records = self.list_records(zone=zone)
+
+        if not self.ex_record_exists(record_id, records):
+            raise RecordDoesNotExistError(value='', driver=self,
+                                          record_id=record_id)
+
+        for record in records:
+            if record_id == record.id:
+                ret_record = record
+
+        return ret_record
+
+    def create_zone(self, zone_id, type='master', ttl=None, extra={}):
```

Comment:
Not a good idea to default to mutable types (dicts, lists), I will change it to default to `None` and do `extra = extra or {}` inside the method body when merging the PR.